src/winbind.c

```diff
@@ -49,28 +49,54 @@ static struct wbcContext *winbind_get_context(void)
     return ctx;
 }
 
-#else /* HAVE_PTHREAD */
+static long signal_ctx = 0;
 
-/* non thread-safe version */
-static struct wbcContext *gctx = NULL;
+#endif /* HAVE_PTHREAD */
+
+void *winbind_get_context(void)
+{
+#ifdef HAVE_PTHREAD
+    const char *envvar;
+
+    /* we return signal_ctx if we are told to use per thread
+     * context, the code knows how to cope correctly */
+    envvar = getenv("GSSNTLMSSP_WB_TLS_CTX");
```

Comment:
Can we implement also an option for compile-time switch of MT behaviour?

We have seen problems with passing environment variables for 'sudo/setcap' processes (privileged processes can suppress intrusion from non-privileged caller). So in our case we possibly prefer to apply some more general solution - like a specific macro during compilation.

Reply:
yes, I think we can do a compile time option